packages/hop-node/src/watchers/BondWithdrawalWatcher.ts

```diff
@@ -347,6 +369,97 @@ class BondWithdrawalWatcher extends BaseWatcher {
+    if (transfersWithExpectedTransferNonce.length > 1) {
+      throw new PreTransactionValidationError(`transferNonce (${txTransferNonce}) exists in multiple transfers in db`)
+    }
+    if (transfersWithExpectedTransferNonce.length === 0) {
+      throw new PreTransactionValidationError(`transferNonce (${txTransferNonce}) does not exist in db`)
+    }
+  }
+
+  async validateLogsWithBackupRpc (txParams: SendBondWithdrawalTxParams): Promise<void> {
+    // Validate logs with backup RPC endpoint, if it exists
+    const calculatedDbTransfer = await this.getCalculatedDbTransfer(txParams)
+    const blockNumber = calculatedDbTransfer.transferSentBlockNumber
+
+    const redundantRpcUrls = getRedundantRpcUrls(this.chainSlug) || []
+    for (const redundantRpcUrl of redundantRpcUrls) {
+      const redundantProvider = new providers.JsonRpcProvider(redundantRpcUrl)
+
+      // TODO: Better way to do this
+      const l2Bridge = contracts.get(this.tokenSymbol, this.chainSlug)?.l2Bridge
+      const events = await l2Bridge.connect(redundantProvider).queryFilter(
+        l2Bridge.filters.TransferSent(),
+        blockNumber,
+        blockNumber
+      )
+      const eventParams = events.filter((x: any) => x.args.transferId === txParams.transferId)[0]
+      if (!eventParams) {
+        throw new PreTransactionValidationError(`TransferSent event not found for transferId ${txParams.transferId} at block ${blockNumber}`)
+      }
+
+      // TODO: better way to do this
+      if (
+        (eventParams.args.transferId !== txParams.transferId) ||
+        (Number(eventParams.args.chainId) !== txParams.destinationChainId) ||
+        (eventParams.args.recipient !== txParams.recipient) ||
+        (eventParams.args.amount.toString() !== txParams.amount.toString()) ||
+        (eventParams.args.transferNonce.toString() !== txParams.transferNonce.toString()) ||
+        (eventParams.args.bonderFee.toString() !== txParams.bonderFee.toString()) ||
+        (eventParams.args.amountOutMin.toString() !== txParams.amountOutMin.toString()) ||
+        (eventParams.args.deadline.toString() !== txParams.deadline.toString()) ||
+        (eventParams.args.index.toString() !== txParams.transferSentIndex.toString())
+      ) {
+        throw new PreTransactionValidationError(`TransferSent event does not match db. eventParams: ${JSON.stringify(eventParams)}, calculatedDbTransfer: ${JSON.stringify(calculatedDbTransfer)}`)
+      }
+    }
+  }
+
+  async getCalculatedDbTransfer (txParams: SendBondWithdrawalTxParams): Promise<Transfer> {
+    const {
+      destinationChainId,
+      recipient,
+      amount,
+      transferNonce,
+      bonderFee,
+      amountOutMin,
+      deadline
+    } = txParams
+
+    const calculatedTransferId = getTransferId(destinationChainId, recipient, amount, transferNonce, bonderFee, amountOutMin, deadline)
+    return this.db.transfers.getByTransferId(calculatedTransferId)
```

Comment:
suggestion: maybe throw `PreTransactionValidationError` error if db item is not found